diff --git a/android/build.gradle b/android/build.gradle
index b106a1a1013dce07ddba5099d359fe8a5408c461..ca255dee1cc913336f27829ac9d407dda58b0670 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -61,8 +61,8 @@ dependencies {
     androidTestImplementation "androidx.test.ext:junit:$androidxJunitVersion"
     androidTestImplementation "androidx.test.espresso:espresso-core:$androidxEspressoCoreVersion"
     implementation "androidx.coordinatorlayout:coordinatorlayout:1.2.0"
-    implementation 'net.zetetic:android-database-sqlcipher:4.5.3'
-    implementation "androidx.sqlite:sqlite:2.4.0"
+    implementation 'net.zetetic:sqlcipher-android:4.17.0@aar'
+    implementation "androidx.sqlite:sqlite:2.6.2"
     //security library
     implementation "androidx.security:security-crypto:1.1.0-alpha06"
     implementation "androidx.biometric:biometric:1.1.0"
diff --git a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java
index 5a477ae4469011ef1f3f1c804edae9eded5d399f..ecce9267bffb1c1e524d6f0b021e4c1b6aea364b 100644
--- a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java
+++ b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java
@@ -26,6 +26,7 @@ import com.getcapacitor.community.database.sqlite.SQLite.ImportExportJson.JsonSQ
 import com.getcapacitor.community.database.sqlite.SQLite.ImportExportJson.UtilsEncryption;
 import com.getcapacitor.community.database.sqlite.SQLite.ImportExportJson.UtilsJson;
 import java.io.File;
+import java.nio.charset.StandardCharsets;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -35,9 +36,8 @@ import java.util.List;
 import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import net.sqlcipher.Cursor;
-import net.sqlcipher.database.SQLiteDatabase;
-import net.sqlcipher.database.SQLiteException;
+import net.zetetic.database.sqlcipher.SQLiteCursor;
+import net.zetetic.database.sqlcipher.SQLiteDatabase;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
@@ -115,7 +115,7 @@ public class Database {
      * Initialize the SQLCipher Libraries
      */
     private void InitializeSQLCipher() {
-        SQLiteDatabase.loadLibs(_context);
+        System.loadLibrary("sqlcipher");
     }

     public SupportSQLiteDatabase getDb() {
@@ -247,7 +247,7 @@ public class Database {
         if (_mode.equals("encryption")) {
             if (_isEncryption) {
                 try {
-                    _uCipher.encrypt(_context, _file, SQLiteDatabase.getBytes(password.toCharArray()));
+                    _uCipher.encrypt(_context, _file, password.getBytes(StandardCharsets.UTF_8));
                 } catch (Exception e) {
                     String msg = "Failed in encryption " + e.getMessage();
                     Log.v(TAG, msg);
@@ -260,7 +260,7 @@ public class Database {
         if (_mode.equals("decryption")) {
             if (_isEncryption) {
                 try {
-                    _uCipher.decrypt(_context, _file, SQLiteDatabase.getBytes(password.toCharArray()));
+                    _uCipher.decrypt(_context, _file, password.getBytes(StandardCharsets.UTF_8));
                     password = "";
                 } catch (Exception e) {
                     String msg = "Failed in decryption " + e.getMessage();
@@ -273,9 +273,9 @@ public class Database {
         }
         try {
             if (!isNCDB() && !this._readOnly) {
-                _db = SQLiteDatabase.openOrCreateDatabase(_file, password, null);
+                _db = SQLiteDatabase.openOrCreateDatabase(_file, password, null, null);
             } else {
-                _db = SQLiteDatabase.openDatabase(String.valueOf(_file), password, null, SQLiteDatabase.OPEN_READONLY);
+                _db = SQLiteDatabase.openDatabase(String.valueOf(_file), password, null, SQLiteDatabase.OPEN_READONLY, null);
             }
             if (_db != null) {
                 if (_db.isOpen()) {
@@ -301,12 +301,6 @@ public class Database {
                         close();
                         _db = null;
                         throw new Exception(msg);
-                    } catch (SQLiteException e) {
-                        String msg = "Failed in setVersion " + e.getMessage();
-                        Log.v(TAG, msg);
-                        close();
-                        _db = null;
-                        throw new Exception(msg);
                     }
                     if (_version > curVersion && _vUpgObject != null && _vUpgObject.size() > 0) {
                         //                           if (_vUpgObject != null && _vUpgObject.size() > 0) {
@@ -973,12 +967,12 @@ public class Database {
      */
     public JSArray selectSQL(String statement, ArrayList<Object> values) throws Exception {
         JSArray retArray = new JSArray();
-        Cursor c = null;
+        SQLiteCursor c = null;
         if (_db == null) {
             return retArray;
         }
         try {
-            c = (Cursor) _db.query(statement, values.toArray(new Object[0]));
+            c = (SQLiteCursor) _db.query(statement, values.toArray(new Object[0]));
             while (c.moveToNext()) {
                 JSObject row = new JSObject();
                 for (int i = 0; i < c.getColumnCount(); i++) {
diff --git a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsDrop.java b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsDrop.java
index 5a91fcf05d5a7766fea78e3e419a815cbb573601..8342a8974a17ac1c8b85e2784a993d7f6167b0b3 100644
--- a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsDrop.java
+++ b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsDrop.java
@@ -6,7 +6,7 @@ import java.util.ArrayList;
 import java.util.Dictionary;
 import java.util.Enumeration;
 import java.util.List;
-import net.sqlcipher.Cursor;
+import net.zetetic.database.sqlcipher.SQLiteCursor;

 public class UtilsDrop {

@@ -20,7 +20,7 @@ public class UtilsDrop {

     public List<String> getTablesNames(Database db) throws Exception {
         List<String> tables = new ArrayList<String>();
-        Cursor cursor = null;
+        SQLiteCursor cursor = null;
         String query = "SELECT name FROM sqlite_master WHERE ";
         query += "type='table' AND name NOT LIKE 'sync_table' ";
         query += "AND name NOT LIKE '_temp_%' ";
@@ -28,7 +28,7 @@ public class UtilsDrop {
         query += "AND name NOT LIKE 'android_%' ";
         query += "ORDER BY rootpage DESC;";
         try {
-            cursor = (Cursor) db.getDb().query(query);
+            cursor = (SQLiteCursor) db.getDb().query(query);
             cursor.moveToFirst();
             while (!cursor.isAfterLast()) {
                 String tableName = cursor.getString(0);
@@ -51,12 +51,12 @@ public class UtilsDrop {

     public List<String> getViewNames(Database db) throws Exception {
         List<String> views = new ArrayList<String>();
-        Cursor cursor = null;
+        SQLiteCursor cursor = null;
         String query = "SELECT name FROM sqlite_master WHERE ";
         query += "type='view' AND name NOT LIKE 'sqlite_%' ";
         query += "ORDER BY rootpage DESC;";
         try {
-            cursor = (Cursor) db.getDb().query(query);
+            cursor = (SQLiteCursor) db.getDb().query(query);
             cursor.moveToFirst();
             while (!cursor.isAfterLast()) {
                 String viewName = cursor.getString(0);
@@ -113,10 +113,10 @@ public class UtilsDrop {

     public List<String> getIndexesNames(Database db) {
         List<String> indexes = new ArrayList<String>();
-        Cursor cursor = null;
+        SQLiteCursor cursor = null;
         String query = "SELECT name FROM sqlite_master WHERE ";
         query += "type='index' AND name NOT LIKE 'sqlite_%';";
-        cursor = (Cursor) db.getDb().query(query);
+        cursor = (SQLiteCursor) db.getDb().query(query);
         cursor.moveToFirst();
         while (!cursor.isAfterLast()) {
             String indexName = cursor.getString(0);
@@ -152,10 +152,10 @@ public class UtilsDrop {

     public List<String> getTriggersNames(Database db) {
         List<String> triggers = new ArrayList<String>();
-        Cursor cursor = null;
+        SQLiteCursor cursor = null;
         String query = "SELECT name FROM sqlite_master WHERE ";
         query += "type='trigger';";
-        cursor = (Cursor) db.getDb().query(query);
+        cursor = (SQLiteCursor) db.getDb().query(query);
         cursor.moveToFirst();
         while (!cursor.isAfterLast()) {
             String triggerName = cursor.getString(0);
diff --git a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLCipher.java b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLCipher.java
index e6f24e34cf8344c8c6b4780b6313917325982b74..1585bc256058b5b323c4341a6837195ae9829ba1 100644
--- a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLCipher.java
+++ b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLCipher.java
@@ -5,9 +5,8 @@ import android.content.SharedPreferences;
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
-import net.sqlcipher.database.SQLiteDatabase;
-import net.sqlcipher.database.SQLiteException;
-import net.sqlcipher.database.SQLiteStatement;
+import net.zetetic.database.sqlcipher.SQLiteDatabase;
+import net.zetetic.database.sqlcipher.SQLiteStatement;

 public class UtilsSQLCipher {

@@ -36,12 +35,12 @@ public class UtilsSQLCipher {
      * @return the detected state of the database
      */
     public State getDatabaseState(Context ctxt, File dbPath, SharedPreferences sharedPreferences, GlobalSQLite globVar) {
-        SQLiteDatabase.loadLibs(ctxt);
+        System.loadLibrary("sqlcipher");
         if (dbPath.exists()) {
             SQLiteDatabase db = null;

             try {
-                db = SQLiteDatabase.openDatabase(dbPath.getAbsolutePath(), "", null, SQLiteDatabase.OPEN_READONLY);
+                db = SQLiteDatabase.openDatabase(dbPath.getAbsolutePath(), "", null, SQLiteDatabase.OPEN_READONLY, null);

                 db.getVersion();

@@ -50,7 +49,7 @@ public class UtilsSQLCipher {
                 try {
                     String passphrase = sharedPreferences.getString("secret", "");
                     if (passphrase.length() > 0) {
-                        db = SQLiteDatabase.openDatabase(dbPath.getAbsolutePath(), passphrase, null, SQLiteDatabase.OPEN_READONLY);
+                        db = SQLiteDatabase.openDatabase(dbPath.getAbsolutePath(), passphrase, null, SQLiteDatabase.OPEN_READONLY, null);
                         db.getVersion();
                         return (State.ENCRYPTED_SECRET);
                     } else {
@@ -59,7 +58,13 @@ public class UtilsSQLCipher {
                 } catch (Exception e1) {
                     try {
                         if (globVar.secret.length() > 0) {
-                            db = SQLiteDatabase.openDatabase(dbPath.getAbsolutePath(), globVar.secret, null, SQLiteDatabase.OPEN_READONLY);
+                            db = SQLiteDatabase.openDatabase(
+                                dbPath.getAbsolutePath(),
+                                globVar.secret,
+                                null,
+                                SQLiteDatabase.OPEN_READONLY,
+                                null
+                            );
                             db.getVersion();
                             return (State.ENCRYPTED_GLOBAL_SECRET);
                         } else {
@@ -92,11 +97,11 @@ public class UtilsSQLCipher {
      * @throws IOException
      */
     public void encrypt(Context ctxt, File originalFile, byte[] passphrase) throws IOException {
-        SQLiteDatabase.loadLibs(ctxt);
+        System.loadLibrary("sqlcipher");

         if (originalFile.exists()) {
             File newFile = File.createTempFile("sqlcipherutils", "tmp", ctxt.getCacheDir());
-            SQLiteDatabase db = SQLiteDatabase.openDatabase(originalFile.getAbsolutePath(), "", null, SQLiteDatabase.OPEN_READWRITE);
+            SQLiteDatabase db = SQLiteDatabase.openDatabase(originalFile.getAbsolutePath(), "", null, SQLiteDatabase.OPEN_READWRITE, null);
             int version = db.getVersion();

             db.close();
@@ -135,7 +140,7 @@ public class UtilsSQLCipher {
     }

     public void decrypt(Context ctxt, File originalFile, byte[] passphrase) throws IOException {
-        SQLiteDatabase.loadLibs(ctxt);
+        System.loadLibrary("sqlcipher");

         if (originalFile.exists()) {
             // Create a temporary file for the decrypted database in the cache directory
@@ -146,7 +151,8 @@ public class UtilsSQLCipher {
                 decryptedFile.getAbsolutePath(),
                 "",
                 null,
-                SQLiteDatabase.OPEN_READWRITE
+                SQLiteDatabase.OPEN_READWRITE,
+                null
             );

             // Open the encrypted database with the provided passphrase
@@ -154,7 +160,8 @@ public class UtilsSQLCipher {
                 originalFile.getAbsolutePath(),
                 new String(passphrase),
                 null,
-                SQLiteDatabase.OPEN_READWRITE
+                SQLiteDatabase.OPEN_READWRITE,
+                null
             );

             int version = encryptedDb.getVersion();
@@ -196,14 +203,14 @@ public class UtilsSQLCipher {
         }
     }

-    public void changePassword(Context ctxt, File file, String password, String nwpassword) throws IOException {
-        SQLiteDatabase.loadLibs(ctxt);
+    public void changePassword(Context ctxt, File file, String password, String nwpassword) throws Exception {
+        System.loadLibrary("sqlcipher");

         if (file.exists()) {
-            SQLiteDatabase db = SQLiteDatabase.openDatabase(file.getAbsolutePath(), password, null, SQLiteDatabase.OPEN_READWRITE);
+            SQLiteDatabase db = SQLiteDatabase.openDatabase(file.getAbsolutePath(), password, null, SQLiteDatabase.OPEN_READWRITE, null);

             if (!db.isOpen()) {
-                throw new SQLiteException("database " + file.getAbsolutePath() + " open failed");
+                throw new Exception("database " + file.getAbsolutePath() + " open failed");
             }
             db.changePassword(nwpassword);
             db.close();
diff --git a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLite.java b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLite.java
index 202d29123d1ef84c23a5d47f263fc0093334ca9e..f1c3d15c0621e2fabf49d56df00d113af2372fb3 100644
--- a/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLite.java
+++ b/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsSQLite.java
@@ -6,7 +6,7 @@ import com.getcapacitor.JSArray;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import net.sqlcipher.Cursor;
+import net.zetetic.database.sqlcipher.SQLiteCursor;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
@@ -19,7 +19,7 @@ public class UtilsSQLite {
         String SELECT_CHANGE = "SELECT total_changes()";
         Boolean success = true;
         int ret = Integer.valueOf(-1);
-        Cursor cursor = (Cursor) db.query(SELECT_CHANGE, null);
+        SQLiteCursor cursor = (SQLiteCursor) db.query(SELECT_CHANGE, null);
         if (cursor != null) {
             if (cursor.moveToFirst()) {
                 ret = Integer.parseInt(cursor.getString(0));
@@ -33,7 +33,7 @@ public class UtilsSQLite {
         String SELECT_CHANGE = "SELECT last_insert_rowid()";
         Boolean success = true;
         long ret = (long) -1;
-        Cursor cursor = (Cursor) db.query(SELECT_CHANGE, null);
+        SQLiteCursor cursor = (SQLiteCursor) db.query(SELECT_CHANGE, null);
         if (cursor.moveToFirst()) {
             ret = Long.parseLong(cursor.getString(0));
         }
